### PR TITLE
tidb-cloud: update a changefeed note to remove outdated info

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Remove "and the TiKV node size is at least 8 vCPU and 16 GiB" from two documents, which is outdated. This change has been confirmed by @grovecai.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/changefeed-sink-to-mysql.md [hunk 1]
```diff
@@ -9,8 +9,7 @@ This document describes how to stream data from TiDB Cloud to MySQL using the **
 
 > **Note:**
 >
-> - To use the changefeed feature, make sure that your TiDB Dedicated cluster version is v6.4.0 or later and the TiKV node size is at least 8 vCPU and 16 GiB.
-> - Currently, TiDB Cloud only allows up to 5 changefeeds per cluster.
```
Comment (lilin90) [inline tidb-cloud/changefeed-sink-to-mysql.md:13]: FYI. Removed this line because this info has also been covered in the following "Restrictions" section.

**Review by grovecai — APPROVED** [review]
LGTM

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**